prow/gerrit/adapter/adapter.go

```diff
@@ -231,12 +233,64 @@ func createCache(cloneURI *url.URL, cookieFilePath string, cacheSize int, config
 	return cache, nil
 }
 
+type Change struct {
+	changeInfo gerrit.ChangeInfo
+	instance   string
+}
+
+func (c *Controller) syncChange(latest client.LastSyncState, changeChan <-chan Change, log *logrus.Entry, wg *sync.WaitGroup) {
+	for changeStruct := range changeChan {
+		change := changeStruct.changeInfo
+		instance := changeStruct.instance
+
+		log := log.WithFields(logrus.Fields{
+			"branch":   change.Branch,
+			"change":   change.Number,
+			"repo":     change.Project,
+			"revision": change.CurrentRevision,
+		})
+
+		cloneURI, err := makeCloneURI(instance, change.Project)
+		if err != nil {
+			log.WithError(err).Error("makeCloneURI.")
+		}
+
+		c.repoCacheMapMux.Lock()
+		cache, ok := c.repoCacheMap[cloneURI.String()]
+		if !ok {
+			if cache, err = createCache(cloneURI, c.cookieFilePath, c.cacheSize, c.configAgent); err != nil {
+				c.repoCacheMapMux.Unlock()
+				wg.Done()
```

Comment:
nit: It would be better to defer this at the beginning of the function rather than calling it before every return point.